### ISSUE TITLE
surf_mom_flux_U/V cannot be horizontally remapped as output

### ISSUE THREAD

**Issue by AaronDonahue** [E3SM-Project/scream#2257]
I found this issue when testing horizontal and vertical remapping of fields using the shoc_standalone test.  I ran into the following error:
```
components/eamxx/src/share/grid/
  remap/coarsening_remapper.cpp:280: FAIL:
  src.get_header().get_identifier().get_layout().rank()>1 || src.get_header().
  get_alloc_properties().get_padding()==0
  Error! We don't support 2d scalar fields that are padded.
```
and tracked the error down to the two fields `surf_mom_flux_U` and `surf_mom_flux_V`.  Note, `surf_mom_flux` by itself works.  Just the U and V subfields cause the error.

**Comment by bartgol** [comment]
I am not sure we do need that restriction, tbh. We do not pack 2d fields usually, so maybe I added that check b/c such a field would have seemed suspicious. But I don't really recall.

**Comment by bartgol** [comment]
Oh, one thing: we should prob check if the field is contiguous, rather than padded. The `surf_mom_flux_U/V` fields are _not_ contiguous, since they're subviewed along an index that is not the slowest striding one. So we cannot pack them. But then, everything should still work fine if we set `can_pack = false` in that case...

**Comment by bartgol** [comment]
This was fixed in #2367
